app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php

```diff
@@ -73,7 +73,7 @@ public function getOptionText($value)
             }
         }
         // End
-        if (isset($options[$value])) {
+        if (is_null($value) === false  &&  is_array($value) === false && isset($options) && isset($options[$value])) {
```

Comment:
Hi @aditisinghcedcoss,

> Array keys can only be scalar, but $value is an array.

It looks like it's enough to check `is_scalar($value)`. Please update code and do amend commit.

Condition like `is_null($value) === false ` looks really odd and is never a good idea.

Reply:
@orlangur sir,

i have made amendmend.

Thank you.

Reply:
@aditisinghcedcoss, only those checks are needed:

```
is_scalar($value) && isset($options[$value])
```

Please squash changes into a single commit and do a force push.

Reply:
@orlangur sir,

I have squashed changes into a single commit and forced pushed.

Reply:
@orlangur sir,

Please Review.

Thank you